content/docs/rfcs/35/README.md

```diff
@@ -0,0 +1,336 @@
+- [`ChaChaPoly`](http://www.noiseprotocol.org/noise.html#the-chachapoly-cipher-functions) for symmetric authenticated encryption (16 bytes block size);
+- [`SHA256`](http://www.noiseprotocol.org/noise.html#the-sha256-hash-function) hash function used in [`HMAC`](http://www.noiseprotocol.org/noise.html#hash-functions) and [`HKDF`](http://www.noiseprotocol.org/noise.html#hash-functions) keys derivation chains (32 bytes output size);
+
+We note that all [design requirements](#Design-requirements) on exchanged messages would be satisfied only *after* a supported Noise handshake is completed, 
+corresponding to a total of 1 Round Trip Time communication *(1-RTT)*.  
+In particular, identity-hiding properties can be guaranteed only if the recommendation described in [After-handshake](#After-handshake) are implemented.
+
+
+In the following, we assume that communicating parties reciprocally know an initial [`contentTopic`](/spec/14/#wakumessage) 
+where they can send/receive the first handshake message.
+We further assume that multiple initiators can start an handshake with the same recipient on an initial `contentTopic`,
+which can be then directly associated to the recipient's identity.
+
+The second handshake message SHOULD be sent/received on a `contentTopic` deterministically derived from the first handshake message (using, for example, `HKDF`).
+This allows
+- the recipient to efficiently continue the handshakes started by each initiator;
+- the initiators to efficiently associate the initiator's second handshake message to their first handshake message.
+
+After the second handshake message is correctly received by initiators, the recommendation described in [After-handshake](#After-handshake) SHOULD be implemented.
+
+### Encryption Primitives
+
+The symmetric primitives supported are:
+- [`ChaChaPoly`](https://www.ietf.org/rfc/rfc7539.txt) for authenticated encryption (16 bytes block size).
+
+## Specification
+
+When [14/WAKU-MESSAGE version](/spec/14/#payload-encryption) is set to 2, 
+the corresponding `WakuMessage`'s `payload` field either contains:
+- a Noise handhshake message and/or a Noise transport message;
+- a `ChaChaPoly` ciphertext.
+
+The following fields are concatenated to form the `payload` field:
+
+ - `protocol-id`: identifies the protocol or primitive in use (1 byte). 
+ Supported values are:
+     - `0`:  protocol specification omitted (set for [after-handshake](#After-handshake) messages);
+     - `10`: Noise protocol `Noise_K1K1_25519_ChaChaPoly_SHA256`;
+     - `11`: Noise protocol `Noise_XK1_25519_ChaChaPoly_SHA256`;
+     - `12`: Noise protocol `Noise_XX_25519_ChaChaPoly_SHA256`;
+     - `13`: Noise protocol `Noise_XXpsk0_25519_ChaChaPoly_SHA256`;
+     - `30`: `ChaChaPoly` symmetric encryption.
+ - `handshake-message-len`: the length in bytes of the Noise handshake message  (1 byte). 
+ If `protocol-id` is not equal to `0`, `10`, `11`, `12`, `13`, this field MUST be set to `0`;
```

Comment:
Isn't it easier to say 
```suggestion
 If `protocol-id` is   equal to `30`  this field MUST be set to `0`;
```

Reply:
The stress is on: if it is not a noise protocol, the handshake message is empty. The other way around, although much shorter and direct, seems to give a special property to chachapoly encryption (and may suggest that it is not zero when protocol-id is not 30, which is not true). I don't have a strong opinion on this by the way.

Reply:
both are fine, I'll leave it up to you